### ISSUE TITLE
[SOLVED] Step-by-step documentation for executing Commands on multiple Tasks, in Columns mode

### ISSUE THREAD

**Issue by RonCam** [bram85/topydo#209]
**I can't locate a link for the  syntax, or for giving an example --[ for using this feature](https://github.com/bram85/topydo/pull/98/commits/ef360c2fd071653940c3f55668b19e9f400b0321) -- in Columns mode.**  
I see the above link appears to give reference to a command, mentioned as both 'repeat' and '. repeats' (same thing?) but searching the [complete topydo web site,](https://www.topydo.org/) for either term, produces 'zero matches'.

Exactly what steps are to be taken to execute multiple commands, while in Columns mode?  **At the moment, attempts to do this by selecting multiple Tasks and then entering a command at the Columns-mode terminal prompt -- are failing.**

I was trying to add the same _due: tag_ to multiple items, for example.  Perhaps I need an example of the correct command syntax to accomplish this?

Actually, I'm not even certain if _repeat/. repeats_ is really a 'Command' but it's the closest thing to one, that I could find in the link.  **A link to documentation, or giving one practical example for what I'm hoping to do, would be most helpful.**

_repeat_ is accepted at the command line, without giving an error but doing nothing.  _help repeat_ brings up _help_ but no reference to repeat can be seen.

**Marking of multple tasks in Columns mode is working fine, but I don't see where to go after doing this ...**
Help, please.

**Comment by RonCam** [comment]
Is the answer that Task ID numbers cannot be used to execute commands on multiple items?  I recall that it a previous question I discovered that "Task ID numbers cannot be used to create a filter expression".  



create a filter expression

**Comment by mruwek** [comment]
I will try to answer that one, as I'm author of the commit referenced above.

> Exactly what steps are to be taken to execute multiple commands, while in Columns mode?

1. Mark the items (by default by hitting **m** key ).
2. Execute a command:
a) by hitting some other [column shortcut](https://www.topydo.org/#ColumnShortcuts). Try for instance **pra** (hit **p**, release, then **r**, release, then **a**). Items previously marked should have priority *A* now.
b) by entering a command with `{}` placeholder where You would normally put todo ID. Try for instance: `pri {} A`.

The second method is mentioned in the [documentation of Command Line in Column UI](https://www.topydo.org/#ColumnUICommandLine)

> I was trying to add the same due: tag to multiple items, for example. 

1. Mark the relevant items.
2. Enter `append {} due:mon` in the command-line widget

> Actually, I'm not even certain if repeat/. repeats is really a 'Command' but it's the closest thing to one, that I could find in the link. A link to documentation, or giving one practical example for what I'm hoping to do, would be most helpful.

It is not. It is an action inside Column UI. It is triggered only with key-shortcut (by default **.** as in VIM). It's mechanism of action is probably best learned by an example:

1. Mark some items.
2. Enter `append {} Lorem Ipsum`
3. Mark some other items
4. Hit **.**
5. Have fun :)

**Comment by mruwek** [comment]
Bottom line: We probably need to record some sort of video demonstrating all of the awesomeness of the topydo. What do You think @bram85 ?

**Comment by RonCam** [comment]
@mruwek , thank you very much.  I haven't yet tried it, but your directions are so clear, that I have no doubt my problem is solved.  I can hardly wait to experience this _additional_ 'awesomeness' in person!! ;-) 

Your examples make all the difference.  I didn't pick-up on how '{}' was to be used, in particular, just from reading the documentation. 

Now that the problem is solved, I shall change the Subject line so that someone searching through these issues for help -- will find your directions, more easily.

**Comment by bram85** [comment]
@RonCam: do you have a suggestion on how to improve the documentation regarding the usage of {} ?

Looking at this issue, I realize that repetition is missing in the example configuration and the documentation. To be honest I forgot about this functionality at all :/

@mruwek: screencasts is something I want to do for a long time but cannot find the time for it (Big Life Events and such). Keeping track of GitHub issues and updating the docs is about the most I can do at the moment. Hopefully in a few months time (but contributions are always welcome).

**Comment by RonCam** [comment]
@bram85: Now that I re-read the relevant part of the Documentation itself, _Todo item actions_ -- _after_ following @mruwek's tutorial -- the Documentation is perfectly accurate.  

My only suggestion would a link, in the section _Todo item actions_, to  [ColumnUICommandLine](https://www.topydo.org/#ColumnUICommandLine), plus a suggestion that the user scroll to the bottom and read: _Run command on marked items_.  There the user will find a use-example for { } and perhaps that could be expanded with one or more of @mruwek's examples, given above.

I don't see how it's possible to insert a tutorial into a documentation section, but for anyone searching _Issues_ for even a few of the words in this thread's Subject line, they will find @mruwek's excellent tutorial, and then all should be clear.

A comment on 'Screencasts', and I assume that means the animated .gif, such as the one titled _Demo_?  I made use of that when I first came across this program, but found to understand what was happening I had to install a browser extension to enable pausing of animated .gifs.  Without this, it ran too fast to be of practical use.  Personally, I get more information out of written material -- with what @mruwek wrote, as the best example.
